llvm/lib/RemoteCachingService/CMakeLists.txt

```diff
@@ -1,27 +1,32 @@
 if (LLVM_CAS_ENABLE_REMOTE_CACHE)
 add_subdirectory(RemoteCacheProto)
 add_subdirectory(RemoteCacheServer)
```

Comment:
Any reason why you decide not to expand these two directories?

Reply:
I may be wrong but because the 

`generate_llvm_protos_library(LLVMRemoteCacheKVProto "compilation_caching_kv.proto" GRPC)
generate_llvm_protos_library(LLVMRemoteCacheCASProto "compilation_caching_cas.proto" GRPC)`

lines are in the `RemoteCacheProto` lib, the header `compilation_caching_cas.grpc.pb.h` is not generated during the compilation step for Client.cpp because the link step comes after, and so I get a build error like:

`/Users/shubham/Development/llvm-project-cas/llvm-project/llvm/lib/RemoteCachingService/RemoteCacheServer/RemoteCacheServer.cpp:12:10: fatal error: 'compilation_caching_cas.grpc.pb.h' file not found
`

I could probably flatten `RemoteCacheServer` in though, but I decided to only flatten the ones that were actually causing an issue

Reply:
Could we keep `RemoteCacheServer` files as a separate library? Only `llvm-remote-cache-test` tool depends on that, nothing else needs to link to that code.

Reply:
Sure, we can keep `RemoteCacheServer` in its old structure.

Maybe it is cleaner if we fold NullService into Service and select implementation using macros but it is not bad in its current status.

Reply:
@akyrtzi done!